src/wildcard.cpp

```diff
@@ -444,7 +444,7 @@ static bool wildcard_test_flags_then_complete(const wcstring &filepath, const wc
     // Compute the description.
     wcstring desc;
     if (expand_flags & expand_flag::gen_descriptions) {
-        desc = file_get_desc(filepath, lstat_res, lstat_buf, stat_res, stat_buf, stat_errno);
+        desc = file_get_desc(filepath, lstat_res, lstat_buf, stat_res, stat_buf, stat_errno, executables_only);
```

Comment:
Can we at least make this its own variable? The control flow here is not very obvious.

So this:

```cpp
bool file_is_executable = false;
if (executables_only) {
    if (is_executable && waccess(filepath, X_OK) == 0)) {
        file_is_executable = true;
    } else {
        return false;
    }
}
```

where we have `if (executables_only && (!is_executable || waccess(filepath, X_OK) != 0))`

Reply:
LGTM then